HeterogeneousCore/CUDAUtilities/test/Launch_t.cpp

```diff
@@ -0,0 +1,31 @@
+#include "HeterogeneousCore/CUDAUtilities/interface/launch.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/requireCUDADevices.h"
+
+
+
+#include <cuda_runtime.h>
+#include<cstdio>
+
+#undef __global__
+#define __global__ inline __attribute__((always_inline))
```

Comment:
this is needed to avoid multiple definition of the same symbol

Reply:
alternative is to have also the c++ definition in its own .cc (not inlined)

Reply:
For this sample case, wouldn't it be enough to have everything in the .cu file ?

Reply:
Sorry, of course that does not work...

What we are trying to do for Cupla and Alpaka is to have the whole implementation in something like `test/implement/Launch_t.cc`,  and then let scram build two versions by having

### `test/Launch_t.cpp`
```c++
#include "implement/Launch_t.cc"
```

### `test/Launch_t.cu`
```c++
#define CUDA_KERNELS_ON_CPU
#include "implement/Launch_t.cc"
```

Reply:
Yes, but the idea of the test is to have a single file compiled by gcc twice (see buildfile)
to test that indeed we can launch kernels from gcc and that the same code will run instead on cpu if ```CUDA_KERNELS_ON_CPU``` is defined (in this case as a compiler option).
of course for cuda we need the additional .cu file to compile  the device code.
For symmetry one can claim that cpu kernels should be compiled in their on cc (as at the end I do in the vertex producer together with a minimal driver).
Still for cpu the code must be forced inlined to avoid multiple symbols.

so in my opinion (at least with this model) 
```
#define __global__ inline __attribute__((always_inline))
```
in case of cpu code will be required (and apparently does not harm cuda code).
This is done in cudaCompact.h. I tried to keep this specific test as self-included as possible.

Reply:
Yes, and that is my previous standard: one .h, one .cc one .cu.
here I tried one .cc and one .cu the latter with ONLY the kernel, no driver code.
I want to test launching from code compiled with gcc (having in mind that both cpu and gpu code shall resides in the same "load units", which is not the case in this test I agree).
I can build two tests (or three) to see what is needed to have both gpu and cpu code compiled, loaded and then run in the same executable (with eventually the driver code compiled by gcc even for the gpu case).